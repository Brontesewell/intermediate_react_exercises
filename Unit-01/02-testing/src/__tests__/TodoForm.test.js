import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TodoForm from '../TodoForm';

let props = {
  handleSubmit() {}
};

test('renders without crashing', () => {
  shallow(<TodoForm {...props} />);
});

test('matches snapshot for new form', () => {
  const tree = renderer.create(<TodoForm {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('matches snapshot for edit form', () => {
  const tree = renderer
    .create(<TodoForm {...props} title="a" description="b" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
