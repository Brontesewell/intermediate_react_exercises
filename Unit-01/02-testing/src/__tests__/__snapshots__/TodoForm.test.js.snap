exports[`test matches snapshot for edit form 1`] = `
<form
  onSubmit={[Function]}>
  <div
    className="input-area">
    <label>
      Title:
      <input
        name="title"
        onChange={[Function]}
        placeholder="What do you need to do?"
        type="text"
        value="a" />
    </label>
  </div>
  <div
    className="input-area">
    <label>
      Description:
      <input
        name="description"
        onChange={[Function]}
        placeholder="Details, please!"
        type="text"
        value="b" />
    </label>
  </div>
  <input
    type="submit"
    value="Edit this todo!" />
</form>
`;

exports[`test matches snapshot for new form 1`] = `
<form
  onSubmit={[Function]}>
  <div
    className="input-area">
    <label>
      Title:
      <input
        name="title"
        onChange={[Function]}
        placeholder="What do you need to do?"
        type="text"
        value="" />
    </label>
  </div>
  <div
    className="input-area">
    <label>
      Description:
      <input
        name="description"
        onChange={[Function]}
        placeholder="Details, please!"
        type="text"
        value="" />
    </label>
  </div>
  <input
    type="submit"
    value="Add this todo!" />
</form>
`;
